src/python/WMCore/WMSpec/StdSpecs/PromptReco.py

```diff
@@ -55,14 +55,24 @@ def buildWorkload(self):
         if self.globalTagConnect:
             scenarioArgs['globalTagConnect'] = self.globalTagConnect
 
-        recoOutMods = self.setupProcessingTask(recoTask, taskType, self.inputDataset,
-                                               scenarioName = self.procScenario,
-                                               scenarioFunc = "promptReco",
-                                               scenarioArgs = scenarioArgs,
-                                               splitAlgo = self.procJobSplitAlgo,
-                                               splitArgs = self.procJobSplitArgs,
-                                               stepType = cmsswStepType,
-                                               forceUnmerged = True)
+        if 'ALCARECO' in self.writeTiers:
+            recoOutMods = self.setupProcessingTask(recoTask, taskType, self.inputDataset,
+                                                   scenarioName = self.procScenario,
+                                                   scenarioFunc = "promptReco",
+                                                   scenarioArgs = scenarioArgs,
+                                                   splitAlgo = self.procJobSplitAlgo,
+                                                   splitArgs = self.procJobSplitArgs,
+                                                   stepType = cmsswStepType,
+                                                   forceUnmerged = [ "write_ALCARECO" ])
```

Comment:
To avoid duplicating this code, how about you do something like:
`forceUnmerged = ["write_ALCARECO"] if 'ALCARECO' in self.writeTiers else False)`